cfme-versions.gemspec

```diff
@@ -1,29 +1,28 @@
-pwd = File.expand_path("../", __FILE__)
-$LOAD_PATH.unshift(pwd) unless $LOAD_PATH.include?(pwd)
-require "cfme-versions"
+
+lib = File.expand_path("../lib", __FILE__)
+$LOAD_PATH.unshift(lib) unless $LOAD_PATH.include?(lib)
+require "cfme/versions/version"
 
 Gem::Specification.new do |spec|
   spec.name          = "cfme-versions"
-  spec.version       = CFME::Versions::version
+  spec.version       = CFME::Versions::GEM_VERSION
   spec.authors       = ["ManageIQ Authors"]
 
   spec.summary       = "Commandline utility and library for viewing versions across CF, CFME, and MIQ"
   spec.description   = "Commandline utility and library for viewing versions across CF, CFME, and MIQ"
   spec.homepage      = "https://github.com/RedHatCloudForms/cfme-versions"
   spec.license       = "Apache-2.0"
 
-  spec.files         = %w[
-    cfme-versions
-    cfme-versions.rb
-    cfme-versions.gemspec
-    README.md
-    LICENSE.txt
-  ]
-
-  spec.bindir        = "."
-  spec.executables   = %w[ cfme-versions ]
-  spec.require_paths = %w[ . ]
+  # Specify which files should be added to the gem when it is released.
+  # The `git ls-files -z` loads the files in the RubyGem that have been added into git.
+  spec.files         = Dir.chdir(File.expand_path('..', __FILE__)) do
+    `git ls-files -z`.split("\x0").reject { |f| f.match(%r{^(test|spec|features)/}) }
```

Comment:
This is literally the worst way to deal with files.

- It requires `git` is one the system for this file to even compile (can be a real PITA on an appliance)
- Includes directories like `bin/` and files like `.travis.yml` which have no business being in the packaged gem
- This is a **one file gem**...

Reply:
Agreed on bin...not sure why that's not excluded by default with `bundle gem`

Reply:
git is only needed during packaging time, so it's not even a thing for appliances.